### PR TITLE
Update conda manager to 3.19

### PR DESCRIPTION
**Pull request by jeanconn**
Update conda to 3.19

### CODE REVIEW

**Comment by taldcroft** [comment]
:+1:

File: pkgs.manifest [hunk 1]
```diff
@@ -2,8 +2,8 @@
 #
 # Python
 #
-Anaconda-64-2.2.0.tar.gz
-Anaconda-32-2.2.0.tar.gz
+Miniconda-64-3.19.0.tar.gz
+Miniconda-32-3.19.0.tar.gz
```
Comment (jeanconn) [inline pkgs.manifest]: If we want to do an inplace update on HEAD flight without reinstalling conda, we'll need to make these Miniconda build directories and the .installed files to support this change.

Comment (taldcroft) [inline pkgs.manifest]: Sounds feasible.

**Comment by jeanconn** [comment]
Should this also include an update to the overall suite to do a "conda install --file pkgs.conda" as part of the "make all_xx" target?  I think, as this new conda version does a much better job with the solver, that this should be relatively fast, and might be safer than doing installs (in the interim) via "conda update ..." .

**Comment by jeanconn** [comment]
I think your :+1: was to the idea.  But now that I've done an incremental update here and a clean build, I think this is a no-op for the code (it doesn't even update the yaml package) and is safe to push.

**Comment by jeanconn** [comment]
Though I am a bit confused by the output of "conda list" in HEAD /proj/sot/ska, as it shows incorrect versions for some of our pip installed pieces.  Hopefully that will be resolved by a new version of conda/pip .

**Comment by jeanconn** [comment]
(specifically, 'conda list' shows incorrect versions of starcheck, chandra_models, and mica now).

**Comment by taldcroft** [comment]
Those packages use `distutils` instead of `setuptools`.  I think that's the problem since only setuptools provides the metadata to let conda know about package version.  (I haven't done a comprehensive check of the others, but I'm guessing that most / all use `setuptools`).

**Comment by taldcroft** [comment]
And when you say "no-op for the code", what does that mean specifically?  What packages change, and what is different now such that it no longer wants to update yaml?

**Comment by taldcroft** [comment]
Can you remind me about what is different here vs. what originally introduced subtle library problems on GRETA?

**Comment by jeanconn** [comment]
Not sure what you mean.   I pinned libpng and qt in master as that was also
a no-op.  That stops any library problems before they happen(ed).

**Comment by jeanconn** [comment]
If by 'subtle library problems' you just mean that matlab is running python
in an environment with its own libz that precludes using libraries with
dependencies on newer libz.

**Comment by taldcroft** [comment]
> I pinned libpng and qt in master

OK now I have discovered https://github.com/sot/skare/commit/ee1ae22530ca635d4e546c9c0a61eb16c41000e1

That is the missing link, and I was expecting to find that in this PR.

**Comment by jeanconn** [comment]
Ah.  Didn't mean for you to need to go off and discover it,  but it didn't
seem related to the intent of this PR or really requiring of its own.

**Comment by jeanconn** [comment]
Below are diffs on the conda packages that we end up with in the new build vs HEAD /proj/sot/ska.  I see the expected updates to conda, conda-env, pip, and setuptools.  I see some build version incremental changes on already-pinned packages (such as ipython and numpy) and I see that unpinned packages have floated a bit.  I think this all looks good.

```
1c1
< # packages in environment at /proj/sot/ska/arch/x86_64-linux_CentOS-5:
---
> # packages in environment at /data/fido/vnow/arch/x86_64-linux_CentOS-5:
3a4
> backports_abc             0.4                      py27_0    
7,9c8,10
< cffi                      1.0.0                    py27_0    
< conda                     3.12.0                   py27_0    
< conda-env                 2.1.4                    py27_0    
---
> cffi                      1.2.1                    py27_0    
> conda                     3.19.0                   py27_0    
> conda-env                 2.4.5                    py27_0    
18c19
< enum34                    1.0.4                    py27_0    
---
> enum34                    1.1.1                    py27_0    
22c23
< ipython                   3.1.0                    py27_0    
---
> ipython                   3.1.0                    py27_1    
28a30
> libgfortran               1.0                           0    
32c34
< libxml2                   2.9.0                         0    
---
> libxml2                   2.9.2                         0    
38c40
< mistune                   0.5.1                    py27_1    
---
> mistune                   0.7.1                    py27_0    
44c46,47
< numpy                     1.9.2                    py27_0    
---
> numpy                     1.9.2                    py27_2    
> openblas                  0.2.14                        3    
53,54c56,57
< ptyprocess                0.4                      py27_0    
< py                        1.4.26                   py27_0    
---
> ptyprocess                0.5                      py27_0    
> py                        1.4.30                   py27_0    
58c61
< pycparser                 2.13                     py27_0    
---
> pycparser                 2.14                     py27_0    
77c80
< setuptools                16.0                     py27_0    
---
> setuptools                18.8.1                   py27_0    
78a82
> singledispatch            3.4.0.3                  py27_0    
87c91
< terminado                 0.5                      py27_0    
---
> terminado                 0.5                      py27_1    
89c93,94
< tornado                   4.1                      py27_0    
---
> tornado                   4.3                      py27_0    
> wheel                     0.26.0                   py27_1  
```

**Comment by taldcroft** [comment]
Probably OK, but I do note that `openblas` is now being included, which I suspect means that the new `numpy` build actually calls different C-routines underneath.  Here's to hoping that Continuum is running all the numpy tests and that those tests have good coverage.  BTW, in the past I have run `numpy.test()` locally and gotten segfaults.  You might try the same, before and after just for kicks.

**Comment by jeanconn** [comment]
The build also adds libgfortran.  I can check to see if the previous numpy build works with the conda 3.19 pieces if we want to lock it down a bit more.

**Comment by jeanconn** [comment]
Both the old and new builds pass the numpy tests in the virtual box.  It might be simpler to stay with the updated numpy build to keep the HEAD and GRETA builds more in sync now that GRETA already has these build updates.

**Comment by taldcroft** [comment]
That works for me.

**Comment by jeanconn** [comment]
I wanted to have a better test setup in place before moving along with this, but GRETA flight 64 already has conda 3.19, and I can't remember why we haven't just pushed this.

**Comment by taldcroft** [comment]
I don't remember either.  I'm good with this update (in concept), so what about the following process for doing this:

- In a dev ska apply the changes in the same way you will for the inplace update of flight Ska.  Make sure package updates are what you expect.
- Run regression test for one of the ACIS thermal load review tools.
- Run starcheck smoke test and make sure output is expected.
- Build a new dev ska starting from Miniconda and do the same regression tests.

**Comment by jeanconn** [comment]
> In a dev ska apply the changes in the same way you will for the inplace
update of flight Ska. Make sure package updates are what you expect.

Ok.  I'll try this.  I'm not sure I can make something just like the
current head flight ska (drift in nonspecified conda packages). Actually,
in might work to just copy head flight Ska to a virtualbox.  Would that be
acceptable if functional?

**Comment by taldcroft** [comment]
Yes, doing the copy would be fine.

**Comment by jeanconn** [comment]
After doing an update of conda to 3.19.0, f I "check" the packages by attempting to reinstall from pkgs.conda I get the package update to numpy that brings in the openblas and libgfortran discussed earlier in this PR.  Those are in GRETA flight now and I had run those numpy tests.  Do we want to just update numpy in HEAD flight to simplify things?

```
[user@localhost skare]$ conda install --dry-run --no-update-dependencies --file pkgs.conda Fetching package metadata: ....
Solving package specifications: ................................................................................................................................
Package plan for installation in environment /proj/sot/ska/arch/x86_64-linux_CentOS-5:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    libgfortran-3.0            |                0         261 KB
    openblas-0.2.14            |                4         3.6 MB
    numpy-1.9.2                |           py27_2         8.0 MB
    ------------------------------------------------------------
                                           Total:        11.8 MB

The following NEW packages will be INSTALLED:

    libgfortran: 3.0-0       
    openblas:    0.2.14-4    

The following packages will be UPDATED:

    numpy:       1.9.2-py27_0 --> 1.9.2-py27_2
```

**Comment by jeanconn** [comment]
And actually, I guess that's a lie.  

```
ska-fido: egrep '(libgfortran|openblas|numpy)' greta_flight_64/conda_list
libgfortran               1.0                           0  
numpy                     1.9.2                    py27_2  
openblas                  0.2.14                        3  
ska.numpy                 0.8                       <pip>
```

GRETA flight has libgfortran 1.0 not 3.0 and openblas build 3 no 4.

**Comment by jeanconn** [comment]
I pushed a change to just freeze it where it is in HEAD flight for now.

**Comment by jeanconn** [comment]
Update process tested on 64 bit VM was:

```
Create build/x86_64-linux_CentOS-5/Miniconda-64-3.19.0/.installed

conda install conda=3.19.0 setuptools=18.8.1 conda-env=2.4.5 pip=7.1.2 yaml=0.1.4
sqlite=3.8.4.1 python=2.7.9 requests=2.6.0 

The following NEW packages will be INSTALLED:
    wheel:      0.29.0-py27_0

The following packages will be UPDATED:
    conda:      3.12.0-py27_0 --> 3.19.0-py27_0
    conda-env:  2.1.4-py27_0  --> 2.4.5-py27_0 
    pip:        6.0.8-py27_0  --> 7.1.2-py27_0 
    setuptools: 16.0-py27_0   --> 18.8.1-py27_0

conda install --dry-run --no-update-dependencies --file pkgs.conda 
Fetching package metadata: ....
Solving package specifications:
............................................................................................................................\
....
# All requested packages already installed.
```

I then checked that this was a no-op

```
cd ~/git/skare
./configure --prefix=/proj/sot/ska
make all_64
```

The diffs from this incremental update skare vs a new install from scratch are as follows.  I think these are all understandable (build updates on packages with no build specifier in pkgs.conda, pkg updates on packages not controlled in pkgs.conda).  I did think the wheel update was required by conda 3.19.0, but it looks like it might not matter.  I'm not sure how much the dependencies are really checked either way in these environments just now:

```
[user@localhost ~]$ diff -u final_update_conda_list new_conda_list 
--- final_update_conda_list     2016-04-29 14:30:51.000000000 -0400
+++ new_conda_list      2016-04-29 14:31:20.000000000 -0400
@@ -1,16 +1,19 @@
-# packages in environment at /proj/sot/ska/arch/x86_64-linux_CentOS-5:
+# packages in environment at /data/aca/new_conda/arch/x86_64-linux_CentOS-5:
 #
 agasc                     0.3.1                     <pip>
 asciitable                0.8.0                     <pip>
 astropy                   1.0.3                np19py27_0  
 autopep8                  1.1.1                     <pip>
+backports                 1.0                      py27_0  
+backports-abc             0.4                       <pip>
 backports.ssl-match-hostname 3.4.0.2                   <pip>
+backports_abc             0.4                      py27_0  
 beautiful-soup            4.3.2                    py27_0  
 beautifulsoup             3.2.1                     <pip>
 beautifulsoup4            4.3.2                     <pip>
 bsddb                     1.0                      py27_0  
 cairo                     1.12.18                       3  
-cffi                      1.0.0                    py27_0  
+cffi                      1.5.2                    py27_1  
 chandra-aca               0.5.1                     <pip>
 chandra-models            0.9                       <pip>
 chandra.cmd-states        0.9.2                     <pip>
@@ -32,7 +35,7 @@
 docutils                  0.12                     py27_0  
 ecdsa                     0.11                     py27_0  
 egenix-mx-base            3.0.0                     <pip>
-enum34                    1.0.4                    py27_0  
+enum34                    1.1.3                    py27_0  
 find-attitude             0.1                       <pip>
 fontconfig                2.11.1                        3  
 freetype                  2.5.2                         1  
@@ -40,18 +43,18 @@
 git-remote-helpers        0.1.0                     <pip>
 h5py                      2.4.0                     <pip>
 hopper                    0.1                       <pip>
-ipython                   3.1.0                    py27_0  
+ipython                   3.1.0                    py27_1  
 ipython-notebook          3.1.0                    py27_1  
 jedi                      0.8.1                    py27_0  
 jinja2                    2.7.3                    py27_1  
 jpeg                      8d                            0  
 jsonschema                2.4.0                    py27_0  
 kadi                      0.12.2                    <pip>
-libffi                    3.0.13                        0  
+libffi                    3.2.1                         0  
 libpng                    1.5.13                        1  
 libsodium                 0.4.5                         0  
 libtiff                   4.0.2                         1  
-libxml2                   2.9.0                         0  
+libxml2                   2.9.2                         0  
 libxslt                   1.1.28                        0  
 llvmlite                  0.2.2                    py27_1  
 lxml                      3.4.2                    py27_0  
@@ -60,7 +63,7 @@
 maude                     0.1                       <pip>
 mercurial                 1.8.4                     <pip>
 mica                      0.5.3                     <pip>
-mistune                   0.5.1                    py27_1  
+mistune                   0.7.2                    py27_0  
 mpi4py                    1.3                      py27_0  
 mpich2                    1.4.1p1                       0  
 mpld3                     0.2                       <pip>
@@ -80,12 +83,12 @@
 pixman                    0.26.2                        0  
 ply                       3.4                      py27_0  
 psutil                    2.2.1                    py27_0  
-ptyprocess                0.4                      py27_0  
-py                        1.4.26                   py27_0  
+ptyprocess                0.5                      py27_0  
+py                        1.4.31                   py27_0  
 py2cairo                  1.10.0                   py27_2  
 pyasn1                    0.1.7                    py27_0  
 pycosat                   0.6.1                    py27_0  
-pycparser                 2.13                     py27_0  
+pycparser                 2.14                     py27_0  
 pycrypto                  2.6.1                    py27_0  
 pyfits                    3.3                       <pip>
 pyflakes                  0.8.1                    py27_0  
@@ -112,6 +115,7 @@
 setuptools                18.8.1                   py27_0  
 sherpa                    4.7                       <pip>
 shiboken                  1.2.1                    py27_0  
+singledispatch            3.4.0.3                  py27_0  
 sip                       4.16.5                   py27_0  
 six                       1.9.0                    py27_0  
 ska-path                  0.1                       <pip>
@@ -136,14 +140,13 @@
 spyder                    2.3.4                    py27_1  
 sqlalchemy                0.9.9                    py27_0  
 sqlite                    3.8.4.1                       1  
-ssl_match_hostname        3.4.0.2                  py27_0  
-starcheck                 11.6                      <pip>
+ssl_match_hostname        3.4.0.2                  py27_1  
 system                    5.8                           2  
 tables                    2.3.1                     <pip>
-terminado                 0.5                      py27_0  
+terminado                 0.5                      py27_1  
 tk                        8.5.18                        0  
-tornado                   4.1                      py27_0  
-wheel                     0.29.0                   py27_0  
+tornado                   4.3                      py27_0  
+wheel                     0.26.0                   py27_1  
 xija                      0.7                       <pip>
 yaml                      0.1.4                         0  
 zeromq                    4.0.4                         0  
```

**Comment by jeanconn** [comment]
Though now I suppose I'm not sure what "Build a new dev ska starting from Miniconda and do the same regression tests." tells us.  

Basically, now that I've seen the package diffs, testing the new dev ska is really testing a tiny new set of packages brought into new miniconda environment, but those tiny updates are not really related to this PR and will likely change again before we do another full build.

**Comment by jeanconn** [comment]
Either way, I get no diffs* in a 4 way comparison of starcheck and psmc_check output between:

- Ska head flight
- Copy of Ska head flight
- Updated copy of ska head flight with conda 3.19
- new ska runtime started with conda 3.19

(* one tiny diff in starcheck output from the Copy to the Updated copy was caused by my *fixing* a bind mount on the virtualbox so that sybase worked).

**Comment by taldcroft** [comment]
OK, that testing looks sufficient to me.  I would be happy going forward with this at a convenient time early next week.

**Pull request merged**